airflow/providers/google/cloud/hooks/compute_ssh.py

```diff
@@ -215,7 +226,22 @@ def get_conn(self) -> paramiko.SSHClient:
             self._authorize_compute_engine_instance_metadata(pubkey)
 
         proxy_command = None
-        if self.use_iap_tunnel:
+        if self.use_iap_tunnel and self.impersonation_chain:
+            proxy_command_args = [
+                'gcloud',
+                'compute',
+                'start-iap-tunnel',
+                str(self.instance_name),
+                '22',
+                '--listen-on-stdin',
+                f'--project={self.project_id}',
+                f'--zone={self.zone}',
+                '--verbosity=warning',
+                f'--impersonate-service-account={self.impersonation_chain}',
+            ]
+            proxy_command = " ".join(shlex.quote(arg) for arg in proxy_command_args)
+
+        elif self.use_iap_tunnel:
```

Comment:
Can you add a new item to `proxy_command_args` variables instead of duplicating so much code?
```python
if self.impersonation_chain:
    proxy_command_args.append(f'--impersonate-service-account={self.impersonation_chain}')
```

Reply:
does it suit you better now ?